src/scatter_buildblock/ScatterSimulation.cxx

```diff
@@ -741,40 +755,23 @@ Succeeded ScatterSimulation::default_downsampling(bool all_images)
     if(!is_null_ptr(activity_image_sptr) && all_images)
     {
         VoxelsOnCartesianGrid<float>* tmp_act = dynamic_cast<VoxelsOnCartesianGrid<float>* >(activity_image_sptr.get());
-        float _zoom_xy =
-                tmp_act->get_voxel_size().x() / tmpl_density->get_voxel_size().x();
-        float _zoom_z =
-                tmp_act->get_voxel_size().z() / tmpl_density->get_voxel_size().z();
+        VoxelsOnCartesianGrid<float>* tmp = tmpl_density->get_empty_copy();
 
-        BasicCoordinate<3,int> new_size = make_coordinate(tmp_act->get_z_size(),
-                                                          tmp_act->get_y_size(),
-                                                          tmp_act->get_x_size());
 
-        zoom_image_in_place(*tmp_act ,
-                            CartesianCoordinate3D<float>(_zoom_z, _zoom_xy, _zoom_xy),
-                            CartesianCoordinate3D<float>(0,0,0),
-                            new_size,
-                            scaling);
+        zoom_image(*tmp, *tmp_act, scaling);
+        activity_image_sptr.reset(new VoxelsOnCartesianGrid<float>(*tmp));
 
         this->remove_cache_for_integrals_over_activity();
     }
 
     if(!is_null_ptr(density_image_sptr) && all_images)
     {
-        VoxelsOnCartesianGrid<float>* tmp_att = dynamic_cast<VoxelsOnCartesianGrid<float>* >(density_image_sptr.get());
-        float _zoom_xy =
-                tmp_att->get_voxel_size().x() / tmpl_density->get_voxel_size().x();
-        float _zoom_z =
-                tmp_att->get_voxel_size().z() / tmpl_density->get_voxel_size().z();
+        VoxelsOnCartesianGrid<float>* tmp_act = dynamic_cast<VoxelsOnCartesianGrid<float>* >(density_image_sptr.get());
+        VoxelsOnCartesianGrid<float>* tmp = tmpl_density->get_empty_copy();
 
-        BasicCoordinate<3,int> new_size = make_coordinate(tmp_att->get_z_size(),
-                                                          tmp_att->get_y_size(),
-                                                          tmp_att->get_x_size());
 
-        zoom_image_in_place(*tmp_att ,
-                            CartesianCoordinate3D<float>(_zoom_z, _zoom_xy, _zoom_xy),
-                            CartesianCoordinate3D<float>(0,0,0),
-                            new_size, scaling);
+        zoom_image(*tmp, *tmp_act, scaling);
+        density_image_sptr.reset(new VoxelsOnCartesianGrid<float>(*tmp));
```

Comment:
this makes yet another copy of a newly allocated image. Surely not needed.

Reply:
This function is a bit weird. 
let, `zoom_image(im1, im2); `
The function as it works will zoom `im2` according to `im1` and replace the content of` im1` with that of `im2`. Without the copying I kept replacing my attenuation image with the initial activity content.

Reply:
I guess you're referring to the previous `zoom_image_in_place`. My current comment is about this line
```
 density_image_sptr.reset(new VoxelsOnCartesianGrid<float>(*tmp));
```
this allocates yet another copy. it should be
```
 density_image_sptr = tmp;
```
as far as I can see (which implies we don't  need the `tmp` variable anyway).